### ISSUE TITLE
feat: Autosharding protocol changes for req-resp protocols

### ISSUE THREAD

**Issue by chair28980** [waku-org/go-waku#676]
This tracks work necessary to change req-resp protocol requirements for pubsub topic fields to be optional in wire specifications. This implies a spec change, changes to clients to use the simplified specs and changes to server-side Waku implementations to interpret client requests with no populated pubsub topics correctly.

specification: @SionoiS

Priority: Not critical for launch (all main benefits of autosharding is already achieved via API changes)

**Comment by chaitanyaprem** [comment]
Refer to https://github.com/waku-org/nwaku/pull/1857/ of nwaku on changes to be made.

**Comment by chaitanyaprem** [comment]
@SionoiS , can you confirm all changes in nwaku and the spec are done for this and are part of above referenced PR or is there something pending?

**Comment by SionoiS** [comment]
> @SionoiS , can you confirm all changes in nwaku and the spec are done for this and are part of above referenced PR or is there something pending?

@chaitanyaprem This work has not been started, #1857 did not change the protocols, only added autosharding to internals.

This issue will track RFCs and nwaku updates.

**Comment by chaitanyaprem** [comment]
> 

As i saw some changes to Filter spec where pubsubtopic has been made optional, was suspecting it was completed. 
Could you please tag relevant nwaku and rfc issues here so that I can also follow as changes happen?

**Comment by chaitanyaprem** [comment]
Refer to https://github.com/waku-org/nwaku/pull/1937 and take it up once this is complete.

**Comment by fryorcraken** [comment]
See https://github.com/waku-org/nwaku/issues/1937#issuecomment-1778308708

**Comment by chaitanyaprem** [comment]
Closing the issue as it is descoped for go-waku.
